GLMakie/src/drawing_primitives.jl

```diff
@@ -2,6 +2,75 @@ using Makie: transform_func_obs, apply_transform
 using Makie: attribute_per_char, FastPixel, el32convert, Pixel
 using Makie: convert_arguments
 
+# TODO: Maybe move this somewhere else?
+function handle_lights(attr::Dict, screen::Screen, lights::Vector{Makie.AbstractLight})
+    # TODO: make this adjustable?
+    max_lights = 64
+    max_params = 5 * 64
+
+    updater = screen.render_tick
+    MAX_PRIORITY = typemax(Int)
+
+    attr[:lights_length] = map(updater, priority = -1000) do _
```

Comment:
Updating on `render_tick` doesn't work with `record`.

Updating (many) lights is kind of complicated in general, because we want to react to updates of each field for each light and also when lights are added or deleted to the lights vector.

Listening to each field individually would require attaching and removing listeners when the lights vector changes. Detecting that would either require some kind of Vector wrapper or an update on draw.

Instead of bothering with all of that I thought it would be easier to just check the lights vector once per frame and update it then. I'm using `render_tick` for that atm, but I guess we don't want that to update in `record` to avoid triggering events?